### PR TITLE
HARVESTER: Avoid "Operation cannot be fulfilled on nodes" error

### PR DESCRIPTION
**Pull request by n313893254**
#### PR Checklist
~- Is this a multi-tenancy feature/bug?
    - [ ] Yes, the relevant RBAC changes are at:~
~- Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?
    - [ ] Yes, the relevant PR is at:~
~- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:~

Avoid "Operation cannot be fulfilled on nodes" error

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2887
- https://github.com/harvester/harvester/issues/3849
- https://github.com/harvester/harvester/issues/3853

### CODE REVIEW

File: pkg/harvester/edit/harvesterhci.io.host/index.vue [hunk 1]
```diff
@@ -398,16 +406,50 @@ export default {
...
+    async actuallySave(url) {
+      if ( this.isCreate ) {
+        url = url || this.schema.linkFor('collection');
+        const res = await this.value.save({ url });
+
+        if (res) {
+          Object.assign(this.value, res);
+        }
+      } else if (!isEqual(this.originValue, this.value)) {
+        await this.value.save();
```
Comment (WuJun2016) [inline pkg/harvester/edit/harvesterhci.io.host/index.vue:452]: Is this function being used? I think it's the same as the method in the mixin.

File: pkg/harvester/edit/harvesterhci.io.host/index.vue [hunk 2]
```diff
@@ -398,16 +406,50 @@ export default {
...
+        } catch (err) {
+          if ((err.status === 409 || err.status === 403) && count < 3) {
+            count++;
+
+            await this.$store.dispatch(`${ inStore }/find`, {
+              type: LONGHORN.NODES,
+              id:   this.longhornNode.id,
+              opt:  { force: true },
+            });
+
```
Comment (WuJun2016) [inline pkg/harvester/edit/harvesterhci.io.host/index.vue:431]: We can use this [method](https://github.com/harvester/dashboard/blob/bd11739fa3257f74c698e65f122f3d6f7b6df975/shell/mixins/create-edit-view/impl.js#L118) to resolve conflicts.

Comment (n313893254) [reply pkg/harvester/edit/harvesterhci.io.host/index.vue:431]: I don't think we have to use that method, I think use this [method](https://github.com/rancher/dashboard/blob/dcc4a73e498df7455ff292ffad46b2781cb81a7a/shell/models/management.cattle.io.project.js#L104) also make sense.

File: pkg/harvester/edit/harvesterhci.io.host/index.vue (hunk 1 above)
Comment (n313893254) [reply pkg/harvester/edit/harvesterhci.io.host/index.vue:452]: Done

**Review by WuJun2016 — APPROVED** [review]
LGTM, thanks

**Review by WuJun2016 — APPROVED** [review]
LGTM, thanks

**Pull request merged**